xbmc/addons/Repository.cpp

```diff
@@ -160,9 +161,9 @@ std::unique_ptr<CRepository> CRepository::FromExtension(CAddonInfo addonInfo, co
 {
   DirList dirs;
   AddonVersion version("0.0.0");
-  AddonPtr addonver;
-  if (CServiceBroker::GetAddonMgr().GetAddon("xbmc.addon", addonver))
-    version = addonver->Version();
+  if (auto api = CServiceBroker::GetAddonMgr().GetAPIVersion("xbmc.addon"))
```

Comment:
Adding an extra set of () silences compiler warnings about unintended assignment in an if statement.